data-processing/append_mortality_data.py

```diff
@@ -0,0 +1,228 @@
+
+        :param vintage: Year for which to retrieve schema
+        :return: Dictonary of starting position and length
+            for each exiting field of that year
+        """
+
+        import json
+        f = open('mort_schema.json', 'rt')
+        j = json.load(f)
+        return j[str(vintage)]
+
+    def transform_to_schema(self, df, schema):
+        """
+        Splits a fixed-with string dataframe into a specified schema
+
+        :param df: Inout dataframne
+        :param schema: Dictionary of starting position and length for each field
+        :return: Dataframe with columns
+        """
+        from pyspark.sql.types import StringType
+        from pyspark.sql.functions import lit
+
+        df = df.select(
+            df.value.substr(schema['state_s'], schema['state_l']).alias('state'),
+            df.value.substr(schema['county_s'], schema['county_l']).alias('county_fips'),
+            # df.value.substr(schema['year_s'], schema['year_l']).alias('year'),
+            df.value.substr(schema['month_s'], schema['month_l']).alias('month'),
+            df.value.substr(schema['day_s'], schema['day_l']).alias('day')
+            # df.value.substr(schema['weekday_s'], schema['weekday_l']).alias('weekday'),
+            # df.value.substr(schema['manner_s'], schema['manner_l']).alias('manner'),
+        )
+
+        df.printSchema()
+
+        df2 = df
+        df2.show(10)
+        return df2
+
+    def get_county_concordance(self, vintage):
+        import math
+        decade = int(math.floor (vintage / 10.0)) * 10
+
+        file = 's3a://data-engineer.club/crosswalk/xw' + str(decade) + '.csv'
+        xwalk = d.spark.read.csv(file, inferSchema=True, header=True)
+        xwalk = xwalk.select(xwalk.fips_state.alias('st_orig'),
+                             xwalk.fips_county.alias('co_orig'),
+                             xwalk.fips_state_1990.alias('st_ref'),
+                             xwalk.fips_county_1990.alias('co_ref'))
+        xwalk = xwalk.withColumn('st_orig', xwalk.st_orig / 10) \
+                     .withColumn('co_orig', xwalk.co_orig / 10) \
+                     .withColumn('st_ref', xwalk.st_ref / 10) \
+                     .withColumn('co_ref', xwalk.co_ref / 10)
+
+        xwalk.show()
+        return xwalk
+
+    def process_year(self, d, vintage):
+
+        from pyspark.sql.types import IntegerType
+        from pyspark.sql import functions as F
+        from pyspark.sql.functions import concat, unix_timestamp, to_date, lit
+        from pyspark.sql.functions import when
+
+        spark = d.spark
+
+        # Read the input file from S3
+        file = 's3a://data-engineer.club/mort/mort' + str(vintage) + '.txt'
+        df = spark.read.text(file)
+        #df.show(20)
+
+        # Extract fields from fixed-width text file
+        field_pos = self.load_field_positions(vintage)
+        df2 = self.transform_to_schema(df, field_pos)
```

Comment:
lets try to find more descriptive variable names than `df1, df2` etc. :smile:

Also, I've found that when I start writing code where I feel the need to start labelling the variables with sequential numbers is sometimes a sign that I need a separate function for each numbered code block.